content/vscode/remote.md

```diff
@@ -0,0 +1,134 @@
+---
+title: มา Dev กันแบบไม่ต้องลง SDK กันเถอะ (Developing inside a Container) 
+showMetadata: true
+editable: true
+showToc: true
+---
+
+
+
+# หากคุณมีปัญหาหรือรู้สึกเบื่อเหนื่อยกับเรื่องเหล่านี้
+
+- เบื่อกับการเตรียม environment ในการ dev หรือต้องมานั่งทำเอกสารเพื่อบอก team ว่าต้องใช้ Nodejs v. ไหน หรือ dotnet sdk version ไหน
+- หรือเปลี่ยนเครื่องคอมใหม่แต่เหนื่อยกับการไล่ลง SDK ที่ต้องใช้ทำงาน
+- อยากลอง SDK ใหม่ๆ แต่ก็กลัวว่าจะมีปัญหากับ project ที่ทำงานอยู่
+- เพื่อนในทีมไม่ได้แม่นเรื่อง Docker แต่อยากให้เขาได้เริ่มใช้งาน
+- อยากให้คนในทีมใช้ extensions ใน vscode ใน project เหมือนๆกัน
+
+# dotnetthailand ขอเสนอ [Remote Development extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+
+สิ่งที่ต้องเตรียมก็มี (เครื่องเปล่าเปิดซิงก็ทำตามได้ หรือมีอะไรลงไปแล้วก็ข้ามๆไป)
+
+- ลง [Docker](https://docs.docker.com/get-docker/) ก่อนเลย
+- แน่นอนลง [VS Code](https://code.visualstudio.com/download)
+- เสร็จแล้วลง [Remote Development extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+- แค่นั้นแหละจบ
+
+# ก่อนไปลุยทำความเข้าใจมันนิดว่ามันทำงานยังไง
+
+![alt text][architecture]
+
+ref: https://code.visualstudio.com/docs/remote/containers
+
+แบบสั้นๆ หลักการคือจะใช้เครื่องเราทำหน้าที่เก็บ code เฉยๆ และใช้ VS Code ที่ลงบนเครื่องทำหน้าที่แค่แสดงผล แต่กระบวนการ build, run อะไรก็ตามจะไปทำงานใน container โดยผ่านการ mount volume ของ code ที่เราทำงาน
+
+ภาษาที่ Support:
+
+- dotnet core 2, 3, 5
+- nodejs 12, 14, 16
+- PHP
+- Golang
+- Java
+- Python
+- Ruby
+- Rust
+- และอีกเพียบไปลองไล่ดูกันเอา
+
+และไม่ใช่แค่สำหรับ dev อย่างเดียวแต่ยังมี container อย่าง azure cli, Docker in Docker, Kubernetes ที่ทำให้เราไม่ต้องมาลงของพวกนี้ในเครื่องอีกต่อไป ใช้ผ่าน container แล้วเขียน script โลด
+
+# แล้ววิธีใช้ทำยังไง
+
+ขอยกตัวอย่างจากการที่ผู้เขียนทำการ folk project web ตัวนี้เอามาเขียนเนื้อหานี่แหละ
+
+1. แน่นอน clone code ลงมา (หรือจะสร้าง folder เปล่าก็ได้)
+2. เปิด VS Code ปกติเราจะสั่งเปิด folder อย่าเพิ่งจิ้มให้กด cmd + P บน Mac (ผู้เขียนลืม Windows ไปแล้ว) แล้วเลือก *Remote-Containers: Open Folder in Container...*
+
+![alt text][open]
```

Comment:
Same as above.
If you don't want to set alternate text, please use empty square bracket.